modules/sfp_countryname.py

```diff
@@ -0,0 +1,323 @@
+                # Get country name from first index of list
+                # Extract only the text part of the country code
+                matchCountry = matchCountries[0].strip(",").strip("'").strip("\"").strip()
+                countries.add(matchCountry)
+        
+        # If any countries are found
+        if len(countries) > 0:
+            return countries
+
+        return None
+            
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return ["IBAN_NUMBER", "PHONE_NUMBER", "AFFILIATE_DOMAIN_NAME",
+                "CO_HOSTED_SITE_DOMAIN", "DOMAIN_NAME", "SIMILARDOMAIN",
+                "AFFILIATE_DOMAIN_WHOIS", "CO_HOSTED_SITE_DOMAIN_WHOIS",
+                'DOMAIN_WHOIS']
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["COUNTRY_NAME"]
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        evttype = "COUNTRY_NAME"
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        myres = list()
+        countryName = None
+
+        # Process the event data based on incoming event type
+        if eventName == "PHONE_NUMBER":
+            countryName = self.detectCountryFromPhone(eventData)
+        elif eventName in ["DOMAIN_NAME", "SIMILARDOMAIN"] or (eventName == "AFFILIATE_DOMAIN_NAME" and self.opts["affiliate"]) or(eventName == "CO_HOSTED_SITE_DOMAIN" and self.opts["coHosted"]):
+            countryName = self.detectCountryFromTLD(eventData)
+        elif eventName == "IBAN_NUMBER":
+            countryName = self.detectCountryFromIBAN(eventData)
+        elif eventName == "DOMAIN_WHOIS" or (eventName == "AFFILIATE_DOMAIN_WHOIS" and self.opts["affiliate"]) or (eventName == "CO_HOSTED_SITE_DOMAIN_WHOIS" and self.opts["coHosted"]):
+            countryNames = self.detectCountryFromWhoIs(eventData)
+
+        # If whois module data source is recevied, multiple country names might be there
+        if 'countryNames' in locals() and "WHOIS" in eventName:
+            if countryNames is None:
+                self.sf.debug("No associated country name found")
+                return None
+
+            for countryName in countryNames:
+                self.sf.debug("Found country name : " + countryName)
+                myres.append(countryName)
+
+                evt = SpiderFootEvent(evttype, countryName, self.__name__, event)
+                if event.moduleDataSource:
+                    evt.moduleDataSource = event.moduleDataSource
+                else:
+                    evt.moduleDataSource = "Unknown"
+                self.notifyListeners(evt)
+            return None
+
+        # Note : Is checking for duplicates needed? 
+        # Since, incoming data is all unique. There is no chance of duplication
+        # Commented for now. 
```

Comment:
See above comment about the duplication check for incoming events. For outgoing, I suggest to perform the check within the Whois loop, since if the same country is mentioned multiple times in a Whois record, you only want to report it once. But if different countries are mentioned, you want each one reported.